orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/QueueShovel.kt

```diff
@@ -34,32 +40,48 @@ class QueueShovel(
   private val queue: Queue,
   private val previousQueue: Queue,
   private val registry: Registry,
-  private val activator: Activator
+  private val activator: Activator,
+  private val config: DynamicConfigService
 ) {
   private val log = LoggerFactory.getLogger(javaClass)
 
   private val pollOpsRateId = registry.createId("orca.nu.shovel.pollOpsRate")
   private val shoveledMessageId = registry.createId("orca.nu.shovel.pushedMessageRate")
   private val shovelErrorId = registry.createId("orca.nu.shovel.pushedMessageErrorRate")
 
-  @Scheduled(fixedDelayString = "\${queue.shovel.poll-frequency.ms:50}")
+  @Scheduled(fixedRate = rateMs)
+  fun migrateIfActive() {
+    if (!isActive()) {
+      return
+    }
+
+    log.info("Actively shoveling from $previousQueue to $queue")
+    val workDuration = Duration.ofMillis(workDurationMs)
+    val start = Instant.now()
+
+    while (Duration.between(start, Instant.now()) < workDuration) {
+      migrateOne()
+      Thread.sleep(50)
```

Comment:
do we still want to "respect" `queue.shovel.poll-frequency.ms`

Reply:
it didn't seem terribly useful to me and was just "snuck" in there via `@Scheduled`. It would make the bean wiring more complicated for little return IMO

Reply:
yeah, that makes sense